auth/aws.go

```diff
@@ -143,51 +148,53 @@ func (a *AWSAuth) GetToken(f *os.File) (string, error) {
 	intermediate := &iamIntermediateResp{}
 	dErr := decoder.Decode(intermediate)
 	if dErr != nil {
-		return "", fmt.Errorf("Error while trying to parse response from Cerberus: %v", err)
+		return fmt.Errorf("Error while trying to parse response from Cerberus: %v", err)
 	}
 
 	// Decode the binary data from base64
 	binaryData, err := base64.StdEncoding.DecodeString(intermediate.AuthData)
 	if err != nil {
-		return "", fmt.Errorf("Invalid authentication data returned from Cerberus: %s", err)
+		return fmt.Errorf("Invalid authentication data returned from Cerberus: %s", err)
 	}
 	input := &kms.DecryptInput{
 		CiphertextBlob: binaryData,
 	}
 	result, err := a.kmsClient.Decrypt(input)
 	if err != nil {
-		return "", fmt.Errorf("Error while decrypting response: %s", err)
+		return fmt.Errorf("Error while decrypting response: %s", err)
 	}
 	r := &api.IAMAuthResponse{}
 	parseErr := json.Unmarshal(result.Plaintext, r)
 	if parseErr != nil {
-		return "", fmt.Errorf("Error while parsing decrypted response: %s", parseErr)
+		return fmt.Errorf("Error while parsing decrypted response: %s", parseErr)
 	}
 	a.token = r.Token
 	// Set the auth header up to make things easier
 	a.headers.Set("X-Vault-Token", r.Token)
 	a.expiry = time.Now().Add(time.Duration(r.Duration) * time.Second)
-	return a.token, nil
+	return nil
 }
 
 // IsAuthenticated returns whether or not the current token is set and is not expired
 func (a *AWSAuth) IsAuthenticated() bool {
 	return len(a.token) > 0 && time.Now().Before(a.expiry)
 }
 
-// Refresh refreshes the current token
+// Refresh refreshes the current token. For AWS Auth, this is just an alias to
+// reauthenticate against the API.
 func (a *AWSAuth) Refresh() error {
 	if !a.IsAuthenticated() {
 		return api.ErrorUnauthenticated
 	}
-	r, err := Refresh(*a.baseURL, a.headers)
-	if err != nil {
-		return err
-	}
-	a.token = r.Data.ClientToken.ClientToken
-	a.expiry = time.Now().Add((time.Duration(r.Data.ClientToken.Duration) * time.Second) - expiryDelta)
-	a.headers.Set("X-Vault-Token", r.Data.ClientToken.ClientToken)
-	return nil
+	// A note on why we are just reauthenticating: You can refresh an AWS token,
+	// but there is a limit (24) to the number of refreshes and the API requests
+	// that you refresh your token on every SDB creation. When doing this in an
+	// automation context, you could surpass this limit. You could not refresh
+	// the token, but it can get you in to a state where you can't perform some
+	// operations. This is less than ideal but better than having an arbitary
+	// bound on the number of refreshes and having to track how many have been
+	// done.
+	return a.authenticate()
```

Comment:
So you may wait till this code is actually in CMS.
The limit is going to be configurable and not guaranteed to be 24.

The limit and the current count will be returned as part of the metadata in a token.